resqpy/olio/weights_and_measures.py

```diff
@@ -58,27 +75,59 @@ def rq_uom(units):
    if ul in ['1E6 m3/d', '1E6 m3/day']: return '1E6 m3/d'
    if units in ['mD.m', 'mD.ft']: return units
    if ul == 'count': return 'Euc'
-   uom_list = properties_data()['uoms']
-   if units in uom_list: return units
    if ul in uom_list: return ul  # dangerous! for example, 'D' means D'Arcy and 'd' means day
    return 'Euc'
 
 
-def rq_uom_list(units_list):
-   """Returns a list of RESQML uom equivalents for units in list."""
+def convert(x, unit_from, unit_to):
+   """Convert value between two compatible units
+   
+   Args:
+      x (numeric or np.array): value(s) to convert
+      unit_from (str): resqml uom
+      unit_to (str): resqml uom
 
-   rq_list = []
-   for u in units_list: rq_list.append(rq_uom(u))
-   return rq_list
+   Returns:
+      Converted value(s)
 
+   """
+   # TODO: robust handling of errors. At present, bad units are treated as "EUC", and will fail silently.
 
-def p_length_unit(units):
```

Comment:
The p_*_units refer to pagoda units – a historic unit handling that got bundled into this module.  It doesn't have anything to do with RESQML, so shouldn't live in resqpy. However, pagoda will fail to run without changes if we strip the functions out of here. I suggest when we put this live we either keep a copy of the old file and move it into the pagoda source code area, or write a wrapper there.

Reply:
A more ambitious project would be to re-write pagoda to be properly based on resqpy – a much bigger undertaking!